pylsp_jsonrpc/endpoint.py

```diff
@@ -35,6 +36,17 @@ def __init__(self, dispatcher, consumer, id_generator=lambda: str(uuid.uuid4()),
         self._client_request_futures = {}
         self._server_request_futures = {}
         self._executor_service = futures.ThreadPoolExecutor(max_workers=max_workers)
+        self._cancelledRequests = set()
+        self._messageQueue = asyncio.Queue()
+
+    async def consume_task(self):
+        log.warning("starting task")
+        loop = asyncio.get_running_loop()
+        while True:
```

Comment:
Is this infinite loop being exited at any point? i.e., what happens at teardown time